// @scopelift/stealth-address-sdk/dist/config/startBlocks.js
export var ERC5564_StartBlocks;
(function (ERC5564_StartBlocks) {
    ERC5564_StartBlocks[ERC5564_StartBlocks["ARBITRUM_ONE"] = 219468264] = "ARBITRUM_ONE";
    ERC5564_StartBlocks[ERC5564_StartBlocks["ARBITRUM_SEPOLIA"] = 24849998] = "ARBITRUM_SEPOLIA";
    ERC5564_StartBlocks[ERC5564_StartBlocks["BASE"] = 15502414] = "BASE";
    ERC5564_StartBlocks[ERC5564_StartBlocks["BASE_SEPOLIA"] = 7552655] = "BASE_SEPOLIA";
    ERC5564_StartBlocks[ERC5564_StartBlocks["HOLESKY"] = 1222405] = "HOLESKY";
    ERC5564_StartBlocks[ERC5564_StartBlocks["MAINNET"] = 20042207] = "MAINNET";
    ERC5564_StartBlocks[ERC5564_StartBlocks["MATIC"] = 57888814] = "MATIC";
    ERC5564_StartBlocks[ERC5564_StartBlocks["OPTIMISM"] = 121097390] = "OPTIMISM";
    ERC5564_StartBlocks[ERC5564_StartBlocks["OPTIMISM_SEPOLIA"] = 9534458] = "OPTIMISM_SEPOLIA";
    ERC5564_StartBlocks[ERC5564_StartBlocks["SEPOLIA"] = 5486597] = "SEPOLIA";
})(ERC5564_StartBlocks || (ERC5564_StartBlocks = {}));
export var ERC6538_StartBlocks;
(function (ERC6538_StartBlocks) {
    ERC6538_StartBlocks[ERC6538_StartBlocks["ARBITRUM_ONE"] = 219468267] = "ARBITRUM_ONE";
    ERC6538_StartBlocks[ERC6538_StartBlocks["ARBITRUM_SEPOLIA"] = 25796636] = "ARBITRUM_SEPOLIA";
    ERC6538_StartBlocks[ERC6538_StartBlocks["BASE"] = 15502414] = "BASE";
    ERC6538_StartBlocks[ERC6538_StartBlocks["BASE_SEPOLIA"] = 7675097] = "BASE_SEPOLIA";
    ERC6538_StartBlocks[ERC6538_StartBlocks["HOLESKY"] = 1222405] = "HOLESKY";
    ERC6538_StartBlocks[ERC6538_StartBlocks["MAINNET"] = 20042207] = "MAINNET";
    ERC6538_StartBlocks[ERC6538_StartBlocks["MATIC"] = 57888814] = "MATIC";
    ERC6538_StartBlocks[ERC6538_StartBlocks["OPTIMISM"] = 121097390] = "OPTIMISM";
    ERC6538_StartBlocks[ERC6538_StartBlocks["OPTIMISM_SEPOLIA"] = 9658043] = "OPTIMISM_SEPOLIA";
    ERC6538_StartBlocks[ERC6538_StartBlocks["SEPOLIA"] = 5538412] = "SEPOLIA";
})(ERC6538_StartBlocks || (ERC6538_StartBlocks = {}));
